power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -198,6 +198,7 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
                 update[entry.index](*this, found->second, pos, sequence_idx_map[entry.index]);
             }
         }
+        cached_sequence_idx_map_.assign(sequence_idx_map.begin(), sequence_idx_map.end());
```

Comment:
`assign` replaces all contents, right? what happens when 2 consecutive permanent updates are called? e.g.

```python
model = PowerGridModel(...)
model.calculate_power_flow(...)  # this initializes topo, y_bus + all math solvers
model.update(...)                # assign seq_idx_map for list A
model.update(...)                # replace seq_idx_map with list B or append list A???
model.calculate_power_flow(...)  # update y_bus elements only for list B or list A and B
```

Reply:
It will update for list B or list A and B, but implicitly. If `update` can be called in such a way, which is unknown to me, logic can be added to accumulate and explicitly update the union of A and B.

Reply:
Resolved in latest commit.